unmarshal.go

```diff
@@ -204,7 +204,7 @@ func (ro *resourceObject) unmarshalFields(v any) error {
 				continue
 			}
 			relDocument, ok := ro.Relationships[name]
-			if !ok {
+			if !ok || (len(relDocument.DataMany) == 0 && relDocument.DataOne == nil) {
```

Comment:
Can we make this a method `document.isEmpty` to make it _slighty_ more obvious that this is a complete check for an empty document.